internal/util/update/resource-merge.go

```diff
@@ -58,6 +59,11 @@ func (u ResourceMergeUpdater) Update(options UpdateOptions) error {
 	}
 	defer os.RemoveAll(updated.AbsPath())
 
+	err := setters.PerformAutoSetters(updated.AbsPath())
```

Comment:
Set the auto-setters in the updated. Then the existing merge logic(which is well tested) will take care of merging it with local.
1. If an auto-setter is newly added in the updated, that will be pulled onto local with the value set.
2. If an auto-setter is set to different value on local, update operation will not modify it.